packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
```

Comment:
Using a maxSize is not a perfect solution if we have many blocks that need to be rerendered at the same our cache may be invalid when it should not (setAttributes is specific per block).
What if in our hook before the current useMemo we have another useMemo that depends on setAttributes and colorCofig and returns an object whose keys are the name and whose value is the setter function? Then when we need the setter we can just retrieve it from that object.

Reply:
That would recreate all setters when only one needs to change.

I can just create the memoized function inside the hook so that every instance gets its own cache, with a `maxSize` of `colorConfigs.length`, and clear it on unmount.